### ISSUE TITLE
Updating navigator throws error #18 but the same options passed to plain Highcharts Js is working fine

### ISSUE THREAD

**Issue by RamWebDev** [highcharts/highcharts#13020]
#### Expected behaviour
Series should get updated when updated options are passed in.
I am using "Highcharts React Wrapper" (https://www.npmjs.com/package/highcharts-react-official) with HighStocks and when using HighCharts wrapped in React Component as controlled component and updating the series or adding new series is giving below mentioned error: Highcharts error #18: www.highcharts.com/errors/18/
However, when the same chart options is copied and tried with plain JS HighStocks example, then it works fine.

#### Actual behaviour
Throws Highcharts error #18: www.highcharts.com/errors/18/ error

#### Live demo with steps to reproduce
https://codesandbox.io/s/highstock-react-wrapper-issue-hry5u

#### Product version
Highcharts 8.0.0
highcharts-react-official 3.0.0

#### Affected browser(s)
Chrome, IE11

**Comment by RamWebDev** [comment]
Another issue noticed was that same chart options when copy pasted in plain JS HighStocks sample in JSFiddle, the formatting is proper with series data coming up at expected height and position Check the below sample of JSFIddle where-in the same chart option is copy pasted from the React example mentioned above and the chart comes up just fine 

https://jsfiddle.net/ramprasads87/eb065qpr/3/

Please refer to the attached screen captures showing React Wrapped HighStocks chart is rendered distorted (one of the series gets rendered in Navigator region). But with the same options copied and tried with plain Js HighStocks renders as expected

![01_HighStocks_PlainJs_SameOptions_RenderedAsExpected](https://user-images.githubusercontent.com/16088067/75260962-6676bc80-5810-11ea-8cc9-332e91577c4a.png)
![01_HighStocks_React_DistortedExample](https://user-images.githubusercontent.com/16088067/75260965-67a7e980-5810-11ea-89ed-8ee40136f856.png)

**Comment by sebastianbochan** [comment]
Hi @RamWebDev 
Thanks for reporting the issue, but you reported this in wrong repo, please use highcharts-react one: https://github.com/highcharts/highcharts-react

**Comment by RamWebDev** [comment]
@sebastianbochan ,
I did raise this issue in Highchart-react repo and @ppotaczek  (https://github.com/ppotaczek) replied back stating that it is Highcharts issue and I will have to report here in this repo.

Please refer to @ppotaczek 's comment in the link below
https://github.com/highcharts/highcharts-react/issues/203#issuecomment-591005014

**Comment by sebastianbochan** [comment]
Hi @RamWebDev,
Thank you for explanation. The reason of closing ticket was wrong description. Now I see that there is reference to native highcharts demo.

Demo:
- http://jsfiddle.net/BlackLabel/6m4e8x0y/4892/

The problem is related with updating navigator in that specific case.

**Comment by sebastianbochan** [comment]
The problem is more complex. After some debugging I've found two problems.

1. Since 8.0.3 we have an error about incorrect parameters in rect path
2. Highcharts #18 error. Let me explain what happened step by step
    2a. Chart is initialized with array of 2 yAxis (first default and second - navigator)
    2b. Three yAxis are created (according to options), but one of them is assigned to navigator. There is a first problem, because we should have 4 yAxis (3 regular and one for navigator).
    2c. The empty series (`navigator.series`) removes navigator and yAxis (which has `index:1`)
      ```
         "navigator": {
            "series": []
          }
       ```
     2d. The series refers to undefined yAxis (which is not regular yAxis, but assigned to removed navigator). As a result we have an error.

TD:LR; The problem is with wrong order of yAxis and missing one yAxis in array. 

**Workaround**
Declare array of yAxis on init.

Demo:
- http://jsfiddle.net/BlackLabel/Lavwt5cx/

Related github issue:
- https://github.com/highcharts/highcharts/issues/9671

**Comment by Nishit-dev** [comment]
Hi team,
Is there any resolution/workaround for this issue? I am facing the same issue in my code. All I am doing is trying to make an API call in afterSetExtremes event and repopulate the series of graph. I do not understand what is the issue here with the indexes of Y axis. I tried removing the 0th element from the Y axis in my afterSetExtremes method but was unable to resolve the issue. Can anyone give a firm solution here? This issue has halted my current development.

**Comment by pawelfus** [comment]
Hi @Nishit-dev 

I recommend to use other API options instead of `chart.update()`. For example `chart.addSeries()`, `chart.addAxis()`, `chart.yAxis[index].remove()` and `chart.series[index].remove()`. If you just need to update dataset, consider using `chart.series[index].setData()`.

**Comment by Nishit-dev** [comment]
Hi @pawelfus 
I am not using any of the options u mentioned above to update my series. Below are the excerpts of my code. Hope It will help you to understand. The problem does not lie with me updating the series, it has something to do with the Y-axis as mentioned above. Also please note that mine is a multigraph i.e there can be multiple series. [{Series 1},{series 2},{series 3}.......]

// Below method is called in x axis events(aftersetExtremes)
function afterSetExtremes(e) {       
        if (e.trigger === 'navigator'){
            const { chart } = e.target;
            chart.yAxis.shift();      //Removing the first element of yAxis(not sure, if it is required as i was just trying to fix the issue)
            props.onNavigatorChanged(e.min,e.max); //Calls API and update a different props which is fed to series.
            //chart.series.setData(props.graphData);
        }    
    }

series: props.graphData, // props.graphData gets updated in the above api call.

**Comment by pawelfus** [comment]
> series: props.graphData, // props.graphData gets updated in the above api call.

Does it mean you use one of our official wrappers? Our wrappers internally use `chart.update()` when props change..

**Comment by Nishit-dev** [comment]
@pawelfus  if you see the last line in my previous comment, props.graphData is nothing but just an array of objects for my series.
Now, props.onNavigatorChanged is my method which calls an API fetch data and update props.graphData. Now As soon as props.graphData is updated my graph will update with new series(Normal React behaviour). 

My question is why am I getting an highcharts error 18  in this case.  Or can you tell me what is the best way to make an API call and update the graph with the new response from API in case of navigator movement i.e dragging and shrinking of navigator.

**Comment by Nishit-dev** [comment]
@pawelfus  Can you please tell me if this is an open bug in highcharts or is there any work around for this? And if it is a bug is there any ETA for this to get resolved?

**Comment by pawelfus** [comment]
Hi @Nishit-dev 

My question was if you use our official wrapper. Because our official wrappers use internally `chart.update()` which is the root of this bug. The workaround is to resign from using `series: props.graphData,` and use Highcharts API to update chart.

Issue is quite complex - I can not provide any ETA for the fix

**Comment by Nishit-dev** [comment]
@pawelfus  Yes I am using  **highcharts-react-official** wrapper. Sorry, I didn't get you the first time.  And when you say I should resign from using  `series: props.graphData,`, you meant only in this particular case (Navigation Movement) or in normal case also(Graph loading). And Do you think the issue will be resolved when I manually setData using highChartsAPI in my aftersetExtremes() method?

**Comment by pawelfus** [comment]
Yes, that should help. 

I'm not an expert when it comes to highcharts-wrapper, but I think you update chart, in general, in a wrong way, see docs:
https://github.com/highcharts/highcharts-react#optimal-way-to-update

If you have any more questions, regarding the Highcharts+React usage, please contact our support: https://www.highcharts.com/blog/support/ (email, forum, stackoverflow). 

PS: You still can workaround posted by Sebastian: https://github.com/highcharts/highcharts/issues/13020#issuecomment-596623150

**Comment by khlieng** [comment]
This might have been fixed by #15121.
